pkg/planner/core/logical_plan_builder.go

```diff
@@ -6502,6 +6502,14 @@ func (b *PlanBuilder) buildByItemsForWindow(
 		}
 		if col, ok := it.(*expression.Column); ok {
 			retItems = append(retItems, property.SortItem{Col: col, Desc: item.Desc})
+			// We need to attempt to add this column because a subquery may be created during the expression rewrite process.
```

Comment:
This 3 lines is same as the under 3 lines from 6515 to 6521. 
So should you please change those code more unified ~

Reply:
Updated. I have reordered the code to make them have same order.